.github/workflows/on_release.yml

```diff
@@ -163,3 +169,94 @@ jobs:
           tags: |
             permitio/opal-client-cedar:latest
             permitio/opal-client-cedar:${{ env.opal_version_tag }}
+
+      - name: Python setup
+        if: ${{ !(github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == 'true') }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11.8'
+
+      # This is the root file representing the package for all the sub-packages.
+      - name: Bump version - packaging__.py
+        if: ${{ !(github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == 'true') }}
+        run: |
+          version_tag=${{ github.event.release.tag_name }}
+          version_tuple=$(echo $version_tag | sed 's/\./, /g')
+          sed -i "s/VERSION = (.*/VERSION = (${version_tuple})/" packages/__packaging__.py
+          cat packages/__packaging__.py
+          git config --local user.email "eli@permit.io"
+          git config --local user.name "elimoshkovich"
+          git add packages/__packaging__.py
+          git commit -m "Bump version to ${version_tag}"
+
+      - name: Cleanup setup.py and Build every sub-packages
+        if: ${{ !(github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == 'true') }}
+        run: |
+          pip install wheel
+          cd packages/opal-common/ ; rm -rf *.egg-info build/ dist/
+          python setup.py sdist bdist_wheel
+          cd ../..
+          cd packages/opal-client/ ; rm -rf *.egg-info build/ dist/
+          python setup.py sdist bdist_wheel
+          cd ../..
+          cd packages/opal-server/ ; rm -rf *.egg-info build/ dist/
+          python setup.py sdist bdist_wheel
+          cd ../..
+
+      # Upload package distributions to the release - All assets in one step
+      - name: Upload assets to release
```

Comment:
Looks good. I hope it works though without having the release step / mentioning what release (maybe it works when the action is `release: published`? But then what about manual trigger?)

Reply:
By "manual trigger" I mean manually executing the action (as it was done till now).
This is still possible because you still have the `workflow_dispatch` - under `on:`, running it manually now makes no sense because there's no corresponding release. So maybe we should disable that option.

Reply:
It's would be a bit problematic, why do we need manually release? There will be right now CI that should take care of this release.
The workflow_dispatch option as it will be from now on it's only if you want to make "dry run" and that's it.
In case this "dry run" not necessary anymore I can disable it, no problem, just let me know.

Reply:
I know it's problematic - that's what I'm trying to say, that you should remove it.

If you want to keep it for "dry runs" (it's for sure not necessary, maybe can be convenient for when debugging stuff) than running it with `dry_run: False` shouldn't be possible.

In summary, you should either:
1. Completely remove the `workflow_dispatch`. 
OR
2. Remove the `dry_run` as an input flag, and just make it always "dry run" when the trigger is manual dispatch.